lib/cache/storages/Storage.ts

```diff
@@ -0,0 +1,5 @@
+export interface Storage {
+  set<V>(key: string, value: V): Promise<void>;
```

Comment:
Would be nice to make the api chainable and return `Promise<this>` instead of `Promise<void>`. Same for `delete` method.